Formula/singularity.rb

```diff
@@ -1,31 +1,46 @@
 class Singularity < Formula
-  desc "Application containers for Linux"
+  desc "Application container and unprivileged sandbox platform for Linux"
   homepage "https://singularity.hpcng.org"
   url "https://github.com/hpcng/singularity/releases/download/v3.8.2/singularity-3.8.2.tar.gz"
   sha256 "996611dec402b4d372b8b9456dd9ec1cb43712d08502e455f38521bd199856d3"
   license "BSD-3-Clause"
+  revision 1
 
   bottle do
     sha256 cellar: :any_skip_relocation, x86_64_linux: "565338bfa4b430b4e3de41de908e91b225e98fdcba64db857f2aedfc1c782be2"
   end
 
+  # No relocation, the localstatedir to find configs etc is compiled into the program
+  pour_bottle? only_if: :default_prefix
+
   depends_on "go" => :build
   depends_on "openssl@1.1" => :build
-  depends_on "libarchive"
+  depends_on "pkg-config" => :build
+  depends_on "libseccomp"
   depends_on :linux
-  depends_on "pkg-config"
   depends_on "squashfs"
-  depends_on "util-linux" # for libuuid
 
   def install
-    system "./mconfig", "--prefix=#{prefix}"
+    inreplace "pkg/util/singularityconf/config.go" do |s|
+      unsquashfs_dir = Formula["squashfs"].bin.to_s
+      s.sub!(/(directive:"mksquashfs path)/, "default:\"#{unsquashfs_dir}/bin\" \\1")
```

Comment:
Do you really want the replacement to read
```
default:"/home/linuxbrew/.linuxbrew/opt/squashfs/bin/bin
```
? In particular, `bin` is duplicated at the end.

Reply:
It looks like, and I expected it too, but in this case: No it doesn't.
Formula["squashfs"], just gives /home/linuxbrew/.linuxbrew/opt/squashfs/ itself.
I tested this on Ubuntu 18.04 and 21.04 and on both, looks here after checking the normal bindirs:

/home/linuxbrew/.linuxbrew/opt/squashfs/bin/

See:
```
strace -f singularity exec library://alpine cat /etc/alpine-release 2>&1 |grep squash
[pid 1444941] newfstatat(AT_FDCWD, "/bin/unsquashfs", 0xc0001af078, 0) = -1 ENOENT (No such file or directory)
[pid 1444941] newfstatat(AT_FDCWD, "/usr/bin/unsquashfs", 0xc0001af148, 0) = -1 ENOENT (No such file or directory)
[pid 1444941] newfstatat(AT_FDCWD, "/sbin/unsquashfs",  <unfinished ...>
[pid 1444941] newfstatat(AT_FDCWD, "/usr/sbin/unsquashfs",  <unfinished ...>
[pid 1444941] newfstatat(AT_FDCWD, "/usr/local/bin/unsquashfs", 0xc0001af628, 0) = -1 ENOENT (No such file or directory)
[pid 1444941] newfstatat(AT_FDCWD, "/usr/local/sbin/unsquashfs", 0xc0001af7c8, 0) = -1 ENOENT (No such file or directory)
[pid 1444941] openat(AT_FDCWD, "/home/linuxbrew/.linuxbrew/opt/squashfs/bin/unsquashfs", O_RDONLY|O_CLOEXEC) = 8
```
On most systems (assuming the biggest user base is maybe Ubuntu-based and uses snap), snap already has a dependency on squashfs-tools, so it will usually just find the installed tool right away (except on servers and in other VMs or containers.)

Reply:
But you don't use `Formula["squashfs"]` (which btw is a `Formula` object, so it being coerced into a path is very unexpected).

What is the output of
```
brew ruby -e 'require "formula"; puts Formula["squashfs"].bin'
```
?

Reply:
Thanks! That looks as it should be: `/home/linuxbrew/.linuxbrew/opt/squashfs/bin`.
I removed the second "/bin", tested, still works.

I've resolved the other points now and the handling of not being relocatable is like a number of other packages do it, so this should be ok too.

For now, all review points should be resolved.